File: Backend/BankingPaymentsApp_API/BankingPaymentsApp_API/Models/ClientUser.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BankingPaymentsApp_API.Models
{
    public class ClientUser : User
    {
        [ForeignKey("Account")]
        public int? AccountId { get; set; }
        public virtual Account? Account { get; set; }
        public List<Beneficiary>? Beneficiaries { get; set; } = new List<Beneficiary>();
        public List<Employee>? Employees { get; set; } = new List<Employee>();
        [Required(ErrorMessage = "Date of Birth is Required!")]
        [DataType(DataType.Date)]
        public DateTime DateOfBirth { get; set; }
        [Required(ErrorMessage = "Address is Required!")]
        public string Address { get; set; } = null!;
        public bool KycVierified { get; set; } = false;
        [ForeignKey("BankUser")]
        public int? BankUserId {  get; set; }
        public virtual BankUser? BankUser { get; set; }
        public virtual ICollection<Document>? Documents { get; set; } = new List<Document>();
    }
}

File: frontend/BankingPaymentsApp_Frontend/src/app/Models/ClientUser.ts
import { Account } from "./Account";
import { BankUser } from "./BankUser";
import { Beneficiary } from "./Beneficiary";
import { Document } from "./Document";
import { Employee } from "./Employee";
import { User } from "./User";

export interface ClientUser extends User {
  accountId?: number;
  account?: Account;
  beneficiaries?: Array<Beneficiary>;
  employees?: Array<Employee>;
  dateOfBirth: string; 
  address: string;
  kycVierified: boolean;
  documents?: Document [];
  bankUserId:number;
  bankUser:BankUser;
}